drivers/soft_spi/soft_spi.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_soft_spi
+ * @{
+ *
+ * @file
+ * @brief       Software SPI implementation
+ *
+ * @author      Markus Blechschmidt <Markus.Blechschmidt@haw-hamburg.de>
+ * @author      Peter Kietzmann     <peter.kietzmann@haw-hamburg.de>
+ */
+
+#include <stdio.h>
+#include <assert.h>
+
+#include "mutex.h"
+#include "periph/gpio.h"
+
+#include "soft_spi.h"
+#include "soft_spi_params.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/**
+ * @brief   Allocate one lock per SPI device
+ */
+static mutex_t locks[sizeof soft_spi_config];
+
+static inline bool soft_spi_bus_is_valid(soft_spi_t bus)
+{
+    unsigned int soft_spi_num = (unsigned int) bus;
+
+    if (sizeof(soft_spi_config) / sizeof(soft_spi_config[0]) < soft_spi_num || soft_spi_num < 0) {
+        return false;
+    }
+    return true;
+}
+
+void soft_spi_init(soft_spi_t bus)
+{
+    DEBUG("Soft SPI init\n");
+    /* This is left empty because the periph_init function will call this for
```

Comment:
What is left empty and what's the relation the the hardware? Please make clear or remove.

Reply:
that was a leftover comment
:heavy_check_mark: